chain/pool/src/lib.rs

```diff
@@ -118,6 +111,80 @@ impl TransactionPool {
     }
 }
 
+pub struct DrainingIterator<'a> {
+    pool: &'a mut TransactionPool,
```

Comment:
The fields in this struct would need docstrings, it is hard to reverse-engineer what they do by reading the code.